src/main/java/io/spokestack/spokestack/SpeechPipeline.java

```diff
@@ -140,6 +140,22 @@ public void deactivate() {
         }
     }
 
+    /**
+     * Add a new listener to receive events from the speech pipeline.
+     * @param listener The listener to add.
+     */
+    public void addListener(OnSpeechEventListener listener) {
+        this.context.addOnSpeechEventListener(listener);
+    }
+
+    /**
+     * Remove a pipeline listener, allowing it to be garbage collected.
+     * @param listener The listener to remove.
+     */
+    public void removeListener(OnSpeechEventListener listener) {
```

Comment:
Do you have a use case in mind for this? I am fine with it for symmetry, but I have not implemented something similar due to lack of a compelling reason. Same question for the `Spokestack` `remove` too.

Reply:
You mean duplicating it in the pipeline when it's already in the context? That's just for convenience and visibility; it's a pain to have to go in and grab the speech context (or, in the case of `Spokestack`, individual modules) to remove listeners—and to know that's where you need to reach to do it.

The main use case for removes in general is what's in the description—if a client app implements a listener as an inner class of an `Activity`, they'll need to remove that listener when the activity gets killed, or they're going to leak memory as the system tries to destroy and create activities but can't garbage collect them.